Source/WebCore/rendering/RenderLayerFilters.cpp

```diff
@@ -141,35 +141,37 @@ IntOutsets RenderLayerFilters::calculateOutsets(RenderElement& renderer, const F
 GraphicsContext* RenderLayerFilters::beginFilterEffect(RenderElement& renderer, GraphicsContext& context, const LayoutRect& filterBoxRect, const LayoutRect& dirtyRect, const LayoutRect& layerRepaintRect, const LayoutRect& clipRect)
 {
     auto expandedDirtyRect = dirtyRect;
-    auto targetBoundingBox = intersection(filterBoxRect, dirtyRect);
-
-    auto outsets = calculateOutsets(renderer, targetBoundingBox);
+    auto outsets = calculateOutsets(renderer, filterBoxRect);
     if (!outsets.isZero()) {
         LayoutBoxExtent flippedOutsets { outsets.bottom(), outsets.left(), outsets.top(), outsets.right() };
         expandedDirtyRect.expand(flippedOutsets);
     }
 
+    auto targetBoundingBox = intersection(filterBoxRect, expandedDirtyRect);
     if (is<RenderSVGShape>(renderer))
         targetBoundingBox = enclosingLayoutRect(renderer.objectBoundingBox());
-    else {
-        // Calculate targetBoundingBox since it will be used if the filter is created.
-        targetBoundingBox = intersection(filterBoxRect, expandedDirtyRect);
-    }
 
     if (targetBoundingBox.isEmpty())
         return nullptr;
 
-    if (!m_filter || m_targetBoundingBox != targetBoundingBox) {
-        m_targetBoundingBox = targetBoundingBox;
+    auto filterInputs = FilterInputs {
+        expandedDirtyRect,
+        filterBoxRect,
+        m_filterScale,
+        m_preferredFilterRenderingModes
+    };
+
+    if (!m_filter || m_filterInputs != filterInputs) {
         // FIXME: This rebuilds the entire effects chain even if the filter style didn't change.
-        m_filter = CSSFilter::create(renderer, renderer.style().filter(), m_preferredFilterRenderingModes, m_filterScale, m_targetBoundingBox, context);
+        m_filter = CSSFilter::create(renderer, renderer.style().filter(), filterInputs.renderingModes, filterInputs.filterScale, filterInputs.dirtyRect, filterInputs.filterBoxRect, context);
```

Comment:
This is a little be confusing change. Above we calculate 
```
auto expandedDirtyRect = dirtyRect;
expandedDirtyRect.expand(flippedOutsets);
auto targetBoundingBox = intersection(filterBoxRect, expandedDirtyRect); 
```
Here we are now passing `filterInputs.filterBoxRect` which is the argument `filterBoxRect` as targetBoundingBox to `CSSFilter::create()`.  What is wrong in passing m_targetBoundingBox to `CSSFilter::create()`?

And we are now passing `filterInputs.dirtyRect` which is `expandedDirtyRect` as the` inflatedDirtyRect` to `CSSFilter::create()` which is in fact the `filterRegion`. These are too many names to remember.